astropy/utils/masked/tests/test_masked.py

```diff
@@ -295,40 +295,42 @@ def setup_class(self):
         super().setup_class()
         # Ensure we have used MaskedQuantity before - just in case a single test gets
         # called; see gh-15316.
-        Masked(Quantity)
+        self.MQ = Masked(Quantity)
 
-    def test_masked_quantity_class_init(self):
-        # TODO: class definitions should be more easily accessible.
+    def test_masked_quantity_getting(self):
+        # First check setup_class (or previous use) defined a cache entry.
         mcls = Masked._masked_classes[self.a.__class__]
```

Comment:
I think this is a more robust test since `__class__` can be overloaded, but `type()` cannot.

```suggestion
        mcls = Masked._masked_classes[type(self.a)]
```

Reply:
In this case, though, I define `a` myself... 

Seems not quite worth running all CI for again (~1 kWh?), but if your other comment ends up needing a change, happy to do this one too.